feat(actions): refactoring + improvement of github actions

gauthierpetetin opened the pull request:
## **Description**
The purpose for these changes is to refactor the github action that checks issues template and adds labels in order to make it more generic and apply it to PRs as well.
Some code duplication has been eliminated as part of this refactoring as well.

[Same PR for Mobile.](https://github.com/MetaMask/metamask-mobile/pull/7567)

## **Manual testing steps**

1. Go on this [test repo](https://github.com/gauthierpetetin-test/repo_test/)
2. Open a PR
3. Wait for ~30s and `external-contributor` label shall appear on the PR you just opened
4. Edit the PR by deleting the whole description
5. Wait for ~30s and the `INVALID-PR-TEMPLATE` shall be added

## **Screenshots/Recordings**

https://recordit.co/5sH8eRHVmZ

### **Before**

NA

### **After**

NA

## **Related issues**

Fixes https://github.com/MetaMask/metamask-extension/issues/21275

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

gauthierpetetin committed:
feat(actions): refactoring + improvement of github actions

The purpose for these changes is to refactor the github action that checks issues template and adds labels in order to make it more generic and apply it to PRs as well. Some code duplication has been eliminated as part of this refactoring as well.

georgewrmarshall reviewed (commented):
This is excellent! As you noted in slack [my test PR failed](https://github.com/gauthierpetetin-test/repo_test/pull/153) so it may need some revision. My experience with GitHub workflows and yml scripts is limited but from a general code perspective it looks good. I would suggest getting a second reviewer familiar with our workflow scripts 

<img width="949" alt="Screenshot 2023-10-23 at 12 08 59 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/780d35ad-292f-46de-a466-1eda8649ddd9">

gauthierpetetin committed:
fix(action): support PRs from forked repo

When PR is opened from a forked repo, by default, the workflow runs in the context of the forked repo which doesn’t have the LABEL_TOKEN secret defined, and results in an error. Replacing pull_request by pull_request_target allows the workflow to run in the context of the base repo rather than the forked repo. So it has access to LABEL_TOKEN secret and it works as expected.

gauthierpetetin committed:
fix(label): change color of external contributor label

gauthierpetetin commented:
Hi @georgewrmarshall , here's a [fix](https://github.com/MetaMask/metamask-extension/pull/21497/commits/c9731e6c210b366452f6ceace55566da04f3f6eb) to your comment regarding PR from forked repo.
- When PR is opened from a forked repo, by default, the workflow runs in the context of the forked repo which doesn’t have the LABEL_TOKEN secret defined, and results in an error.
- Replacing `pull_request` by `pull_request_target` allows the workflow to run in the context of the base repo rather than the forked repo. So it has access to LABEL_TOKEN secret and it works as expected.

gauthierpetetin commented on .github/scripts/shared/label.ts:
```diff
@@ -0,0 +1,121 @@
+
+export const externalContributorLabel: Label = {
+  name: 'external-contributor',
+  color: 'B60205',
```
Fixed in this [commit](https://github.com/MetaMask/metamask-extension/pull/21497/commits/aabd64c41917b818b324a295855863d0641f98d7)

gauthierpetetin committed:
fix(script): rename file to adopt kebab-case

georgewrmarshall reviewed (approved):
LGTM. Fantastic work 💯  Tested labelling on test repo working well! https://github.com/gauthierpetetin-test/repo_test/pull/159
